dnsprobe/dnsprobe.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const defaultPort = "53"
+
+func main() {
+	if len(os.Args) != 3 {
+		_, _ = fmt.Fprintf(os.Stderr, "Usage: %s <name> <nameserver[:port]>\n", os.Args[0])
+		os.Exit(1)
+	}
+
+	host, portStr, err := net.SplitHostPort(os.Args[2])
+	if err != nil {
+		// TODO: This is a very bad way to check for an error, but `net.SplitHostPort` does not return a named error.
+		// Manually checking if os.Args[2] contains a port is harder than it seems due to having to handle v6 addresses.
+		if strings.Contains(err.Error(), "missing port in address") {
+			host = os.Args[2]
+			portStr = defaultPort
+		} else {
+			log.Fatalf("Cannot parse host[:port] from %q: %v", os.Args[2], err)
+		}
+	}
+
+	port, err := strconv.ParseInt(portStr, 10, 16)
+	if err != nil {
+		log.Fatalf("Cannot parse port %q: %v", portStr, err)
+	}
+
+	resolver := &net.Resolver{
+		PreferGo:     true,
+		StrictErrors: true,
+		Dial: func(ctx context.Context, _, _ string) (net.Conn, error) {
+			return net.DialUDP("udp", nil, &net.UDPAddr{
+				IP:   net.ParseIP(host),
+				Port: int(port),
+			})
+		},
+	}
+
+	// TODO: Make timeout configurable.
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```

Comment:
Based on the test failures on slower emulated architectures, we may need a higher default here. Or make it configurable.

```
4589156005-sut-1     | 2023/04/02 14:17:08 Error looking up "dnssec.works": lookup dnssec.works on 127.0.0.11:53: write udp 127.0.0.1:55071->127.0.0.1:5053: i/o timeout
```

Reply:
Interesting, I wasn't expecting emulation to take such a large hit. Let me see if I can do some sane flag parsing without adding external dependencies.

Reply:
@klutchell I've added a `--timeout` flag and specified a whopping 15s timeout for the test, LMK if that works :)

Those are on different commits for easier reviewability, I can squash them later if you wish.